src/react-native/04-intro-to-jsx/intro-to-jsx.md

```diff
@@ -10,17 +10,177 @@
 
-### Concept TODO
+### What is JSX?
 
-TODO
+JSX is used by React Native developers to define the user interface.
+
+JSX is a special syntax designed to look almost identical to HTML. Developers define the UI using JSX. React transforms that JSX into the HTML the browser displays. JSX can be written alongside standard JavaScript, which makes for a powerful programming environment.
+
+#### Basic JSX looks like HTML
+
+```tsx
+const greeting = <Text>Hello, world!</Text>
+```
+
+This code snippet creates a simple JSX element: a `<Text>` header with the text “Hello, world!” This is similar to writing HTML, but it’s JSX inside a JavaScript file.
+
+### Embedding a JavaScript expression in JSX
+
+```tsx
+const name = "Alice"
+const greeting = <Text>Hello, {name}!</Text>
+```
+
+Here, we embed a JavaScript expression `{name}` within JSX. The value of the name variable is displayed within the `<Text>` tag.
+
+### Combining JSX with standard JavaScript
+
+```tsx
+function welcome(name) {
+  return <Text>Hello, {name}</Text>
+}
+
+const welcomeMessage = welcome("Alice")
+```
+
+This example illustrates a regular JavaScript function being called with a single argument. The `welcome` function returns JSX, showing how JSX can be seamlessly integrated within standard JavaScript functions.
+
+### JSX transpiled
+
+React Native has a procedural `React.createElement` syntax, but most applications do not use it directly. Instead, views are defined and maintained in JSX and will automatically be transpiled into the equivalent `React.createElement` calls at build-time.
+
+```tsx
+function welcome(name) {
+  return React.createElement("Text", null, `Hello, ${name}!`)
+}
+
+const welcomeMessage = welcome("Alice")
+```
+
+This transformation is handled by tools like Vite during the build process, allowing us to write more readable and maintainable code using JSX.
+
+### Differences between JSX and HTML
+
+JSX looks a lot like HTML, and that’s intentional. Browsers understand HTML, JavaScript, and CSS. Eventually, anything we build has to be converted into one of those 3 syntaxes. Since our UI code will eventually end up as HTML, using JSX means it will be easier to understand and debug the end result in the browser.
+
+However, JSX is converted into JavaScript and therefore is not an exact mirror of HTML. Some of the most noticeable differences include:
+
+#### HTML has attributes, JSX has props
+
+This nomenclature difference is because they are technically different, though they might appear to be the same visually.
+
+When talking about the HTML below, you might say that the `href` attribute is a URL:
+
+```html
+<a href="https://www.bitovi.com/academy/">Bitovi Academy</a>
+```
+
+The above is valid JSX too, but we would say that the `href` _prop_ is being passed into the anchor element.
```

Comment:
#### XML has attributes, JSX has props

In XML, attributes are used to provide additional information about elements, and they are always treated as strings. This makes XML straightforward but less flexible in terms of integrating dynamic content directly within the markup.

For example, consider an XML element representing a user:

```xml
<user age="37" isActive="true" name="Jane Doe"></user>
```

JSX allows props (short for properties) to be much more than just strings. Props in JSX can be any JavaScript expression, including any other JavaScript primitive, object, function, etc.

```jsx
const userElement = (
  <User age={37} isActive={true} name={"Mary Jackson"} />
);
```

In the code above, the `age`, `isActive`, and `name` props are all passed into the `User` component.